pkg/catalog/manager/sync.go

```diff
@@ -70,6 +41,139 @@ func (m *Manager) Sync(key string, obj *v3.Catalog) error {
 		return nil
 	}
 
+	cmt := &CatalogManagerType{
+		catalog:        catalog,
+		projectCatalog: nil,
+		catalogType:    client.CatalogType,
+	}
+
 	logrus.Infof("Updating catalog %s", catalog.Name)
-	return m.traverseAndUpdate(repoPath, commit, catalog)
+	return m.traverseAndUpdate(repoPath, commit, cmt)
+}
+
+func (m *Manager) PrjCatalogSync(key string, obj *v3.ProjectCatalog) error {
```

Comment:
nit: can we use Project in the method name? as we are not abbreviating the catalog